### DISCUSSION TITLE
How to declare image resource?

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#12235] [Q&A]
I would like to test changing control theme variants (light/dark) with my custom controls. However, apart from regular brushes etc I would also need to replace images with their inverse colour counterpart. Can I do that via DynamicResource? If so, how do I declare a resource that I can later bind to Image control?

**Comment by danielmayost** [comment]
I think it's wrong to do that, resources are loaded into memory and stay there (even if it's lazy loaded, it ends up being loaded and not released), not something you want in images.

Additionally I think this is technically not possible, unless you want to use juggles (and still no URI):
```
<Application.Resources>
  <Bitmap x:Key="btmp">
    <x:Arguments>
      <sys:String>your/absolute/path</sys:String>
    </x:Arguments>
  </Bitmap>
</Application.Resources>
```

**Reply by lukaszossowski** [reply]
Those are very small icons I'm talking about, so memory usage is negligible. I know what I want could be done via PathIcon (as it is for example with combobox arrow in fluent theme) and then its Foreground, but converting those images to svg paths will be complicated.

WPF has something like this:

`<BitmapImage x:Key="MyImageSource" UriSource="../Media/Image.png" />`

but Avalonia doesn't seem to.

**Reply by danielmayost** [reply]
>  but Avalonia doesn't seem to.

You are probably right.

but it is still possible through code.

**Reply by lukaszossowski** [reply]
I would like to do it entirely through xaml, just like Fluent Theme and Simple theme are done. This is something of an exercise, so I want to do it the proper way.
